src/main/java/seedu/recipe/ui/AddRecipeForm.java

```diff
@@ -0,0 +1,286 @@
+package seedu.recipe.ui;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import javafx.fxml.FXML;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.layout.FlowPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import seedu.recipe.commons.core.LogsCenter;
+
+/**
+ * Represents the form element for users to add {@code Recipe}s
+ */
+public class AddRecipeForm extends UiPart<Region> {
+    private static final String FXML = "RecipeForm.fxml";
+
+    @FXML
+    private TextField nameField;
+
+    @FXML
+    private TextField durationField;
+
+    @FXML
+    private TextField portionField;
+
+    @FXML
+    private TextField tagsField;
+
+    @FXML
+    private FlowPane tags;
+
+    @FXML
+    private VBox ingredientsBox;
+
+    @FXML
+    private VBox stepsBox;
+
+    @FXML
+    private Button saveButton;
+
+    @FXML
+    private Button cancelButton;
+
+    //Logic executors and system logging
+    private final Logger logger = LogsCenter.getLogger(getClass());
+
+    private StringBuilder commands;
+
+    @FunctionalInterface
+    interface CustomFocusChangeListener {
+        void onFocusChange(boolean newValue);
+    }
+
+    /**
+     * Creates a new AddRecipeForm instance.
+     *
+     * @param stringBuilder The {@code StringBuilder} that stores the command string.
+     */
+    public AddRecipeForm(StringBuilder stringBuilder) {
```

Comment:
i personally think `stringBuilder` here should be renamed to `data` cause that more accurately represents what it does, but this is a minor nitpick

(same for the `stringBuilder` in `AddFormCommand`)